static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -103,18 +105,23 @@ function useReplayData({
   const [errors, setErrors] = useState<ReplayError[]>([]);
   const [replayRecord, setReplayRecord] = useState<ReplayRecord>();
 
+  const projectSlug = useMemo(() => {
+    if (!replayRecord) {
+      return null;
+    }
+    return projects.projects.find(p => p.id === replayRecord.project_id)?.slug;
+  }, [replayRecord, projects.projects]);
+
   // Fetch every field of the replay. We're overfetching, not every field is used
   const fetchReplay = useCallback(async () => {
-    const response = await api.requestPromise(
-      `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/`
-    );
+    const response = await api.requestPromise(makeFetchReplayApiUrl(orgSlug, replayId));
     const mappedRecord = mapResponseToReplayRecord(response.data);
     setReplayRecord(mappedRecord);
     setState(prev => ({...prev, fetchingReplay: false}));
-  }, [api, orgSlug, projectSlug, replayId]);
+  }, [api, orgSlug, replayId]);
 
   const fetchAttachments = useCallback(async () => {
-    if (!replayRecord) {
+    if (!replayRecord || !projectSlug) {
```

Comment:
Why do we fetch attachments where there is no project slug?

Reply:
its not that we were fetching attachments without a project slug, its that before we already had the projectSlug from the URL so we never needed to assert against it.

now the project slug comes after we resolve a replayRecord so we have to wait for that first

Reply:
Oops sorry, missed the return below this 🤦